src/Core/Managed/Shared/Extensibility/Implementation/SimpleMetricStatisticsAggregator.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using System.Threading;
+
+    /// <summary>
+    /// Represents aggregator for a single time series of a given metric.
+    /// </summary>
+    internal class SimpleMetricStatisticsAggregator
```

Comment:
Just a thought: having two aggregator classes, `MetricAggregator` and `SimpleMetricStatisticAggregator` feels confusing. What if they were called `Metric` and `MetricAggregator` respectively? As a potential benefit, this would allow to simplify the `IMetricProcessor` definition to `Track(Metric, double value)`, which would also be more resilient to future changes in the `Metric` class.